packages/client/tiptap/extensions/table/Table.tsx

```diff
@@ -0,0 +1,153 @@
+import * as DropdownMenu from '@radix-ui/react-dropdown-menu'
+import TiptapTable, {TableOptions} from '@tiptap/extension-table'
+import {
+  NodeViewContent,
+  NodeViewWrapper,
+  ReactNodeViewRenderer,
+  type NodeViewProps
+} from '@tiptap/react'
+import {useState} from 'react'
+import PlainButton from '../../../components/PlainButton/PlainButton'
+import {cn} from '../../../ui/cn'
+import {twStyled} from '../../../ui/twStyled'
+import AddColumnLeft from './icons/AddColumnLeftSVG'
+import AddColumnRight from './icons/AddColumnRightSVG'
+import AddRowAbove from './icons/AddRowAboveSVG'
+import AddRowBelow from './icons/AddRowBelowSVG'
+import Backspace from './icons/BackspaceSVG'
+import EditTableSVG from './icons/EditTableSVG'
+import Toolbar from './icons/ToolbarSVG'
+
+const Item = twStyled(DropdownMenu.Item)`
+flex items-center gap-2 text-slate-700 hover:bg-slate-200 rounded px-2 py-1 cursor-pointer text-sm
+`
+
+type Highlight =
+  | 'before'
+  | 'after'
+  | 'above'
+  | 'below'
+  | 'row'
+  | 'column'
+  | 'cell'
+  | 'header'
+  | null
+
+function Component(props: NodeViewProps) {
+  const {editor, node} = props
+
+  const isActive = () => {
+    const state = editor.view.state
+    const from = state.selection.from
+    const pos = state.doc.resolve(from)
+
+    for (let i = pos.depth; i >= 0; --i) {
+      if (pos.node(i) === node) {
+        return true
+      }
+    }
+    return false
+  }
+
+  const selected = isActive()
+
+  const onOpenChange = (open: boolean) => {
+    if (open) {
+      editor.chain().focus().setCellAttribute('active', true).run()
+    } else {
+      editor.chain().focus().setCellAttribute('active', null).run()
+    }
+  }
+
+  const [highlight, setHighlight] = useState<Highlight>(null)
+
+  const focus = (what: Highlight) => () => setHighlight(what)
+  const blur = () => setHighlight(null)
+
+  return (
+    <NodeViewWrapper className='relative' data-highlight={highlight}>
+      <NodeViewContent as='table' {...props} />
```

Comment:
-1, not all the props are HTML props, e.g. `getPos`. this is causing errors in the console.